geode-management/src/main/java/org/apache/geode/management/api/ClientAsyncOperationResult.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.management.api;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.springframework.web.client.RestTemplate;
+
+import org.apache.geode.annotations.Experimental;
+
+@Experimental
+public class ClientAsyncOperationResult<V extends JsonSerializable>
+    extends AsyncOperationResult<V> {
+  private RestTemplate restTemplate;
+  private String uri;
+
+  public ClientAsyncOperationResult(RestTemplate restTemplate, String uri) {
+    this.restTemplate = restTemplate;
+    this.uri = uri;
+  }
+
+  @SuppressWarnings("unchecked")
+  /**
+   * this should be the only method to make the request to the locator to check the status of the
+   * operation
+   */
+  private ClusterManagementOperationResult<V> requestStatus() {
+    return restTemplate
+        .getForEntity(uri, ClusterManagementOperationResult.class)
+        .getBody();
+  }
+
+  @Override
+  public boolean cancel(boolean mayInterruptIfRunning) {
+    return false;
+  }
+
+  @Override
+  public boolean isCancelled() {
+    return false;
+  }
+
+  @Override
+  public boolean isDone() {
+    ClusterManagementOperationResult<V> result = requestStatus();
```

Comment:
I was thinking about a way to avoid making the extra trip to the server to get the completed result. We could save the result here as a member variable to get it after isDone returns true.

Reply:
updated to cache the first "done" result as suggested